tt_metal/include/tt_metal/types.hpp

```diff
@@ -53,11 +74,22 @@ class BufferHandle {
     std::shared_ptr<v0::Buffer> buffer_ptr;
 };
 
-// Not likely going to be opaque, but pending review of
-// completion of the prototype of the runtime args.
-class Event;
-class RuntimeArgs;
-class RuntimeArgsData;
+class EventHandle {
+   public:
+    explicit EventHandle(const std::shared_ptr<v0::Event> &event_ptr) noexcept : event_ptr{event_ptr} {}
+    explicit EventHandle(std::shared_ptr<v0::Event> &&event_ptr) noexcept :
+        event_ptr{static_cast<std::shared_ptr<v0::Event> &&>(event_ptr)} {}
+
+    explicit operator const std::shared_ptr<v0::Event> &() const noexcept { return event_ptr; }
+
```

Comment:
Is the plan to swap this from v0::Event to v1::Event as part of removing the reach arounds?

Reply:
I think the plan is to internalize `Event` entirely. Now that I'm looking at the v1 API with fresh eyes, I think the `EnqueueRecordEvent` should be a creation function for `EventHandle` rather than accepting one as an argument. Internally, it initializes it so that its completion can be tracked, so it makes sense for the function to create it rather than the user.